addons/osio-addon/src/main/java/io/fabric8/launcher/osio/client/OsioWitClient.java

```diff
@@ -104,7 +108,18 @@ public void createCodeBase(final String spaceId, final String stackId, final URI
         final Request request = newAuthorizedRequestBuilder(getWitUrl(), "/api/spaces/" + spaceId + "/codebases")
                 .post(create(parse("application/json"), payload))
                 .build();
-        httpClient.executeAndConsume(request, r -> validateCodeBaseResponse(spaceId, repositoryCloneUri, r));
+        httpClient.executeAndConsume(request, r -> {
```

Comment:
Use `httpClient.executeAndParseJson` instead

Reply:
If I want to preserve the logic of `validateCodeBaseResponse` aka `isCodeBaseValid` I can't really do that, can I?

Reply:
Then use `executeAndMap` if you need the `Response`

Reply:
It has a flow anyway, thanks for pointing out - revamping it.